src/AuthProvider.jsx

```diff
@@ -0,0 +1,36 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+const AuthContext = createContext();
+
+export const useAuth = () => {
+    return useContext(AuthContext);
+};
+
+export const AuthProvider = ({ children}) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [userData, setUserData] = useState({ firstName: "", lastName: "", role: "" })
+    const isToken = sessionStorage.getItem('isToken');
```

Comment:
Conventionally you want to save the "is" prefix exclusively for boolean values.

Reply:
A boolean value is stored.
![istoken](https://github.com/user-attachments/assets/ac8c6fe0-c78c-4fd6-a697-b1aad33f101d)

Reply:
Good call. You're right. I noticed later that you are indeed passing around a boolean. But you'll want to cache your token instead.